src/panels/lovelace/cards/hui-area-card.ts

```diff
@@ -35,26 +43,32 @@ import {
   subscribeEntityRegistry,
 } from "../../../data/entity_registry";
 import { forwardHaptic } from "../../../data/haptics";
-import { ActionHandlerEvent } from "../../../data/lovelace";
 import { SubscribeMixin } from "../../../mixins/subscribe-mixin";
 import { HomeAssistant } from "../../../types";
-import { actionHandler } from "../common/directives/action-handler-directive";
-import { toggleEntity } from "../common/entity/toggle-entity";
 import "../components/hui-warning";
 import { LovelaceCard, LovelaceCardEditor } from "../types";
 import { AreaCardConfig } from "./types";
 
-const SENSOR_DOMAINS = new Set(["sensor", "binary_sensor"]);
+const SENSOR_DOMAINS = ["sensor"];
+
+const ALERT_DOMAINS = ["binary_sensor"];
 
-const SENSOR_DEVICE_CLASSES = new Set([
-  "temperature",
-  "humidity",
-  "motion",
-  "door",
-  "aqi",
-]);
+const DEVICE_CLASSES = {
+  sensor: ["temperature", "humidity"],
+  binary_sensor: ["motion"],
+};
 
-const TOGGLE_DOMAINS = new Set(["light", "fan", "switch"]);
+const TOGGLE_DOMAINS = ["light", "switch", "fan"];
+
+const DOMAIN_ICONS = {
```

Comment:
We have `domainIcon` in `src/common/entity/domain_icon.ts` why not use that?

Reply:
Because we dont have a single entity and domain icon expects that

Reply:
Only for fan we could use `domainIcon` or we have to create fake stateObjects and stuff...